### PR TITLE
MULE-17284: Remove unnecesary parentComponetModel parameter passed from

### PR DESCRIPTION
**Pull request by elrodro83**
BeanDefinitionFactory#resolveComponentRecursively

### CODE REVIEW

File: modules/spring-config/src/main/java/org/mule/runtime/config/internal/dsl/spring/BeanDefinitionFactory.java [hunk 1]
```diff
@@ -316,18 +313,13 @@ private void processMuleSecurityManager(ComponentModel componentModel, BeanDefin
...
-  private void resolveComponentBeanDefinition(ComponentModel parentComponentModel, SpringComponentModel componentModel) {
+  private void resolveComponentBeanDefinition(SpringComponentModel componentModel) {
     Optional<ComponentBuildingDefinition<?>> buildingDefinitionOptional =
         componentBuildingDefinitionRegistry.getBuildingDefinition(componentModel.getIdentifier());
     if (buildingDefinitionOptional.isPresent() || customBuildersComponentIdentifiers.contains(componentModel.getIdentifier())) {
-      this.componentModelProcessor.processRequest(new CreateBeanDefinitionRequest(parentComponentModel, componentModel,
+      this.componentModelProcessor.processRequest(new CreateBeanDefinitionRequest(componentModel,
                                                                                   buildingDefinitionOptional.orElse(null)));
     } else {
-      boolean isWrapperComponent = isWrapperComponent(componentModel.getIdentifier(), of(parentComponentModel.getIdentifier()));
```
Comment (gsfernandes) [inline modules/spring-config/src/main/java/org/mule/runtime/config/internal/dsl/spring/BeanDefinitionFactory.java]: Why did you remove this code? If I'm not wrong I saw this error message when an application is missing a mule-plugin dependency and still has usages in its config.

**Comment by elrodro83** [comment]
--validate
